src/main/scala/translating/BAPToIR.scala

```diff
@@ -30,7 +33,7 @@ class BAPToIR(var program: BAPProgram, mainAddress: Int) {
       for (p <- s.out) {
         out.append(p.toIR)
       }
-      val procedure = Procedure(s.name, Some(s.address), blocks, in, out)
+      val procedure = Procedure(s.name, Some(s.address), blocks, in, out, externalFunctionNames.contains(s.name))
```

Comment:
Since identifying the external functions isn't something we get from BAP, it should happen later on when the program is represented by the IR, not during the process of translating to the IR.

Reply:
Removed, although its very inefficient to do this.